main.c

```diff
@@ -118,11 +118,26 @@ static bool surface_is_opaque(struct swaylock_surface *surface) {
 	return (surface->state->args.colors.background & 0xff) == 0xff;
 }
 
+struct zxdg_output_v1_listener _xdg_output_listener;
+static const struct wl_callback_listener surface_ready_listener;
+
 static void create_layer_surface(struct swaylock_surface *surface) {
 	struct swaylock_state *state = surface->state;
 
 	surface->image = select_image(state, surface);
 
+	static bool has_printed_zxdg_error = false;
```

Comment:
It would be better to print this error in `main`. That way we wouldn't need this static variable.